src/rider/main/kotlin/me/fornever/avaloniarider/AvaloniaMessages.kt

```diff
@@ -0,0 +1,90 @@
+package me.fornever.avaloniarider
+
+import com.jetbrains.rider.util.idea.application
+import com.jetbrains.rider.util.idea.getComponent
+import me.fornever.avaloniarider.me.fornever.avaloniarider.AvaloniaIncomingMessage
+import me.fornever.avaloniarider.me.fornever.avaloniarider.AvaloniaOutgoingMessage
+import java.util.*
+
+class AvaloniaMessages {
+    companion object {
+        fun getInstance(): AvaloniaMessages = application.getComponent()
+    }
+
+    val incomingTypeRegistry = mutableMapOf<UUID, Class<*>>()
+    val outgoingTypeRegistry = mutableMapOf<Class<*>, UUID>()
+
+    init {
+        val declaredMessageTypes = AvaloniaMessage::class.nestedClasses
```

Comment:
`nestedClasses` doesn't collect any kind of "derived classes". I think you're just misunderstanding what the "nested class" is. [Here're the docs](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect/-k-class/nested-classes.html). And here's the sample for clarity:

```kotlin
class Foo() {
  class Bar() // not derived from Foo, but nested
  class Baz() : Foo() // derived from Foo and also nested
}
class Frob() : Foo() // derived from Foo but not tested 
```

Hope it clears the confusion. And, yeah, using Kotlin 1.3 and `sealedSubclasses` API would be very good here. Do you need any help with upgrading the Kotlin version?

Reply:
Yes, now I see. Didn't see the trick of putting other classes as nested in the example I used.
So know I switched to Kotlin 1.3.21 and used sealedSubclasses in reflection to fill registry maps. It works correctly. https://github.com/ForNeVeR/AvaloniaRider/pull/7/commits/8297e0c7f768b3d1c328bd1a21992b4ee168d69f